src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -0,0 +1,172 @@
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    // The VS LSP client supports streaming using IProgress<T> on various requests.
+    // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+    [ExportLspMethod(LSP.Methods.TextDocumentReferencesName), Shared]
+    internal class FindAllReferencesHandler : IRequestHandler<LSP.ReferenceParams, object[]>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FindAllReferencesHandler(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object[]> HandleRequestAsync(Solution solution, ReferenceParams referenceParams, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            // The VS LSP client supports streaming using IProgress<T> on various requests.
+            // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+            var locations = ArrayBuilder<LSP.Location>.GetInstance();
+            var document = solution.GetDocumentFromURI(referenceParams.TextDocument.Uri);
+            if (document == null)
+            {
+                return locations.ToArrayAndFree();
+            }
+
+            var findUsagesService = document.Project.LanguageServices.GetService<IFindUsagesLSPService>();
+            var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(referenceParams.Position), cancellationToken).ConfigureAwait(false);
+            var context = new SimpleFindUsagesContext(cancellationToken);
+
+            await findUsagesService.FindReferencesAsync(document, position, context).ConfigureAwait(false);
+
+            if (clientCapabilities?.HasVisualStudioLspCapability() == true)
+            {
+                return await GetReferenceItemsAsync(referenceParams, context, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                return await GetLocationsAsync(referenceParams, context, cancellationToken).ConfigureAwait(false);
```

Comment:
is it actually important to support this case?

Reply:
After talking with David, it was apparently included in order to support Liveshare backwards compatibility, but since back-compat is already broken, I was told it's probably ok to remove for now if an assert was included.